packages/kit/src/core/postbuild/prerender.js

```diff
@@ -41,362 +61,351 @@ function normalise_error_handler(log, input, format) {
-		const index = key.indexOf('#');
-		const path = key.slice(0, index);
-		const id = key.slice(index + 1);
+	if (response.status === 200) {
+		mkdirp(dirname(dest));
 
-		const hashlinks = actual_hashlinks.get(path);
-		// ignore fragment links to pages that were not prerendered
-		if (!hashlinks) continue;
+		log.info(`${response.status} ${decoded}`);
+		writeFileSync(dest, body);
+		written.add(file);
 
-		if (!hashlinks.includes(id)) {
-			handle_missing_id({ id, path, referrers: Array.from(referrers) });
+		if (is_html) {
+			prerendered.pages.set(decoded, {
+				file
+			});
+		} else {
+			prerendered.assets.set(decoded, {
+				type
+			});
 		}
+
+		prerendered.paths.push(decoded);
+	} else if (response_type !== OK) {
+		handle_http_error({ status: response.status, path: decoded, referrer, referenceType });
 	}
 
-	/** @type {string[]} */
-	const not_prerendered = [];
+	manifest.assets.add(file);
+	saved.set(file, dest);
+}
 
-	for (const [route_id, prerender] of prerender_map) {
-		if (prerender === true && !prerendered_routes.has(route_id)) {
-			not_prerendered.push(route_id);
+for (const entry of config.prerender.entries) {
+	if (entry === '*') {
+		for (const [id, prerender] of prerender_map) {
+			if (prerender) {
+				if (id.includes('[')) continue;
+				const path = `/${get_route_segments(id).join('/')}`;
+				enqueue(null, config.paths.base + path);
+			}
 		}
+	} else {
+		enqueue(null, config.paths.base + entry);
 	}
+}
 
-	if (not_prerendered.length > 0) {
-		throw new Error(
-			`The following routes were marked as prerenderable, but were not prerendered because they were not found while crawling your app:\n${not_prerendered.map(
-				(id) => `  - ${id}`
-			)}\n\nSee https://kit.svelte.dev/docs/page-options#prerender-troubleshooting for info on how to solve this`
-		);
+await q.done();
+
+// handle invalid fragment links
+for (const [key, referrers] of expected_hashlinks) {
+	const index = key.indexOf('#');
+	const path = key.slice(0, index);
+	const id = key.slice(index + 1);
+
+	const hashlinks = actual_hashlinks.get(path);
+	// ignore fragment links to pages that were not prerendered
+	if (!hashlinks) continue;
+
+	if (!hashlinks.includes(id)) {
+		handle_missing_id({ id, path, referrers: Array.from(referrers) });
 	}
+}
+
+/** @type {string[]} */
+const not_prerendered = [];
 
-	return { prerendered, prerender_map };
+for (const [route_id, prerender] of prerender_map) {
+	if (prerender === true && !prerendered_routes.has(route_id)) {
+		not_prerendered.push(route_id);
+	}
 }
+
+if (not_prerendered.length > 0) {
+	throw new Error(
+		`The following routes were marked as prerenderable, but were not prerendered because they were not found while crawling your app:\n${not_prerendered.map(
+			(id) => `  - ${id}`
+		)}\n\nSee https://kit.svelte.dev/docs/page-options#prerender-troubleshooting for info on how to solve this`
+	);
+}
+
+// prerender_map is updated during prerendering, so we need to write it back
```

Comment:
I was confused for a moment because "write it back" sounds like it's updated, but the file is generated here and then read later somewhere else. Maybe "...so we need to store it for later retrieval"?

Reply:
This is an old comment, it refers to the fact that `analyse.js` previously wrote `prerender_map` which later got augmented. Now that it writes `metadata.routes` instead, `prerender_map` is entirely generated inside `prerender.js` so it's just writing it, not writing it _back_